auto_nag/scripts/vacant_triage_owner.py

```diff
@@ -0,0 +1,167 @@
+                    continue
+
+                manager = team_managers.get_team_manager(component["team_name"])
+
+                info = {
+                    "id": component["id"],
+                    "manager": manager["name"],
+                    "team": component["team_name"],
+                    "product": product["name"],
+                    "component": component["name"],
+                    "triage_owner": triage_owner,
+                    "status": inactive_users[triage_owner],
+                }
+
+                vacant_components.append(info)
+                self.add(manager["mozilla_email"], info)
+
+        return vacant_components
+
+    def get_email(self, date, bug_ids=[]):
+        data = self.identify_vacant_components()
+        if not data:
+            return None, None
+
+        extra = self.get_extra_for_template()
+        env = Environment(loader=FileSystemLoader("templates"))
+        template = env.get_template(self.template())
+        message = template.render(
+            date=date,
+            data=data,
+            extra=extra,
+            str=str,
+            enumerate=enumerate,
+            plural=utils.plural,
+            no_manager=self.no_manager,
+            table_attrs=self.get_config("table_attrs"),
+            preamble=self.preamble(),
+        )
+        common = env.get_template("common.html")
+        body = common.render(
+            message=message, query_url=utils.split_long_url(self.query_url)
+        )
+
+        return self.get_email_subject(date), body
+
+    def prepare_mails(self):
+        """Prepare nag emails"""
+
+        if not self.data:
+            return []
+
+        template = self.nag_template()
+        if not template:
+            return []
+
+        extra = self.get_extra_for_nag_template()
+        env = Environment(loader=FileSystemLoader("templates"))
+        template = env.get_template(template)
+        mails = []
+        for manager, info in self.data.items():
+            # The same bug can be several times in the list
+            # because we send an email to a team.
+            added_bug_ids = set()
+
+            data = []
+            To = sorted(info.keys())
+            for person in To:
+                data += [
+                    bug_data
+                    for bug_data in info[person]
+                    if bug_data["id"] not in added_bug_ids
+                ]
+                added_bug_ids.update(bug_data["id"] for bug_data in info[person])
+
+            body = template.render(
+                date=self.nag_date,
+                extra=extra,
+                plural=utils.plural,
+                enumerate=enumerate,
+                data=data,
+                nag=True,
+                table_attrs=self.get_config("table_attrs"),
+                nag_preamble=self.nag_preamble(),
+            )
+
+            m = {"manager": manager, "to": set(To), "body": body}
+            mails.append(m)
+
+        return mails
```

Comment:
can we reuse some code from bzcleaner or nag_me? If not, can we define these functions (or part of them) in a common files between them to avoid duplicating the code?

Reply:
This code is mostly taken from `BzCleaner` and `Nag` with some edits to work with components instead of bugs.

I was planning to discuss this with you. I think it needs some refactoring, I have notes for stuff that needs to be refactored (this case is covered). However, I leave that we should have some tests before doing any refactoring to make sure all tools will keep working as intended. We could use mocks similar to libmozdata.

Reply:
If it is simple to extract in a separate function that both BzCleaner, Nag, and this script call, then you can do it as part of this PR.
If it is a bit more complex, you could file a follow-up issue to do it.

Reply:
I created #1345 as a follow-up issue.